Java V2 Add Sitewise Basics Scenario (#6871)
awsdocs/aws-doc-sdk-examples: basics_scenarios/sitewise_scenario/SPECIFICATION.md

# AWS IoT SiteWise Service Scenario Specification

## Overview
This SDK Basics scenario demonstrates how to interact with AWS IoT SiteWise using an AWS SDK. 
It demonstrates various tasks such as creating a SiteWise Asset Model, creating an asset, 
sending data to the asset, reading data, and so on.  Finally this scenario demonstrates how 
to clean up resources. Its purpose is to demonstrate how to get up and running with 
AWS IoT SiteWise and an AWS SDK.

## Resources
This Basics scenario requires an IAM role that has permissions to work with AWS IoT 
SiteWise service. The scenario creates this resource using a CloudFormation template.

## Hello AWS IoT SiteWise

> This program is intended for users not familiar with the AWS IoT SiteWise Service to easily get up and running. The logic is to show use of 

Reviewer: ```suggestion
This program is intended for users not familiar with the AWS IoT SiteWise Service to easily get up and running. The program uses a `listAssetsPaginator` to demonstrate how you can read through Asset information.
```

Author: fixed

`listAssetsPaginator`.

## Basics Scenario Program Flow
The AWS IoT SiteWise Basics scenario executes the following operations.

1. **Create an AWS SiteWise Asset Model**:
   - Description: This step creates an AWS SiteWise Asset Model by invoking the `createAssetModel` method.
   - Exception Handling: Check to see if a `ResourceAlreadyExistsException` is thrown. 

>      If it is thrown, get the asset model ID and move on.

Author: fixed

2. **Create an AWS IoT SiteWise Asset**:
   - Description: This operation creates an AWS SiteWise asset.
   - The method `createAsset` is called to obtain the asset ID.

>    - Exception Handling: Check to see if a `ResourceNotFoundException` is thrown. If so, 

Reviewer: A `ResourceNotFoundException` seems an odd Exception for a create* operation. 

A `[ResourceAlreadyExistsException](https://sdk.amazonaws.com/java/api/latest/software/amazon/awssdk/services/iotsitewise/model/ResourceAlreadyExistsException.html) make more sense.

Author: Nice catch - it is a ResourceNotFoundException.

Author: ResourceALre4adyExists not applicable for this service operation. It will make dup asset names.

Author: Fixed

   display the message and end the program.

3. **Retrieve the property ID values**:
   - Description: To send data to an asset, we need to get the property ID values for the 
   the model properties. This scenario uses temperature and humidity properties.

>    - The method `listAssetModelProperties()` is called to retrieve the property ID values.

Reviewer: Either remove all `()` for methods or add them to all. Right now it's inconsistent.

Author: I removed them from spec description sections.

   - Exception Handling: Check to see if an `IoTSiteWiseException` is thrown. There are not 
   many other useful exceptions for this specific call. If so, display the message and end the program.

4. **Send data to an AWS IoT SiteWise Asset**:
   - Description: This operation sends data to an IoT SiteWise Asset.
   - This step uses the method `batchPutAssetPropertyValue()`.
   - Exception Handling: Check to see if a `ResourceNotFoundException` is thrown. If so, display the message and end the program.

5. **Retrieve the value of the IoT SiteWise Asset property**:
   - Description: This operation gets data from the asset.
   - This step uses the method `getAssetPropertyValue()`.
   - Exception Handling: Check to see if a `ResourceNotFoundException` is thrown. If so, display the message and end the program.

6. **Create an IoT SiteWise Portal**:
   - Description: This operation creates an IoT SiteWise portal.
   - The method `createPortal` is called.

>    - Exception Handling: Check to see if an `IoTSiteWiseException` is thrown.

Reviewer: What to do if the exception is thrown? Move on? or end the program?

Author: Fixed

7. **Describe the Portal**:
   - Description: This operation describes the portal and returns a URL for the portal.
   - The method `describePortal()` is called and returns the URL.
   - Exception Handling: Check to see if a `ResourceNotFoundException` is thrown. If so, display the message and end the program.

> 8. **Create an IoTSiteWise Gateway**:

Reviewer: ```suggestion
8. **Create an IoT SiteWise Gateway**:
```

Author: fixed

   - Description: This operation creates an IoT SiteWise Gateway.
   - The method `createGateway` is called.

>    - Exception Handling: Check to see if an `IoTSiteWiseException` is thrown.

Reviewer: What to do if the exception is thrown? Move on? or end the program?

Author: Fixed

Delete stack requested ....
Stack deleted successfully.
This concludes the AWS SiteWise Scenario
--------------------------------------------------------------------------------

```

## SOS Tags

The following table describes the metadata used in this Basics Scenario.

| action                         | metadata file                     | metadata key                            |
|--------------------------------|-----------------------------------|---------------------------------------- |
| `describeGateway`              | iot_sitewise_metadata.yaml        | iotsitewise_DescribeGateway             |
| `deleteGateway `               | iot_sitewise_metadata.yaml        | iotsitewise_DeleteGateway               |
| `createGateway `               | iot_sitewise_metadata.yaml        | iotsitewise_CreateGateway               |
| `describePortal`               | iot_sitewise_metadata.yaml        | iotsitewise_DescribePortal              |
| `listAssetModels`              | iot_sitewise_metadata.yaml        | iotsitewise_ListAssetModels             |
| `deletePortal`                 | iot_sitewise_metadata.yaml        | iotsitewise_DeletePortal                |

> | `createPortals`                | iot_sitewise_metadata.yaml        | iotsitewise_CreatePortal                |

Reviewer: ```suggestion
| `createPortal`                | iot_sitewise_metadata.yaml        | iotsitewise_CreatePortal                |
```

Author: fixed

| `deleteAssetModel`             | iot_sitewise_metadata.yaml        | iotsitewise_DeleteAssetModel            |
| `deleteAsset`                  | iot_sitewise_metadata.yaml        | iotsitewise_DeleteAsset                 |
| `describeAssetModel`           | iot_sitewise_metadata.yaml        | iotsitewise_DescribeAssetModel          |
| `getAssetPropertyValue`        | iot_sitewise_metadata.yaml        | iotsitewise_GetAssetPropertyValue       |
| `batchPutAssetPropertyValue`   | iot_sitewise_metadata.yaml        | iotsitewise_BatchPutAssetPropertyValue  |
| `createAsset`                  | iot_sitewise_metadata.yaml        | iotsitewise_CreateAsset                 |

> | `createAsset`                  | iot_sitewise_metadata.yaml        | iotsitewise_CreateAsset                 |

Reviewer: `createAsset` listed twice.

Author: Fixed
